consensus/hotstuff/signature.go

```diff
@@ -0,0 +1,60 @@
+package hotstuff
+
+import (
+	"github.com/onflow/flow-go/crypto"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// RandomBeaconReconstructor collects verified signature shares, and reconstructs the
+// group signature with enough shares.
+type RandomBeaconReconstructor interface {
+	// TrustedAdd a beacon share without verifying it.
+	TrustedAdd(signerID flow.Identifier, sig crypto.Signature) (bool, error)
+	HasSufficientShares() bool
+	// Reconstruct the beacon signature if there are enough shares. It errors if not enough shares were collected.
+	Reconstruct() (crypto.Signature, error)
+}
+
+// RandomBeaconAggregator aggregates the random beacon signatures
+type RandomBeaconAggregator interface {
+	// TrustedAdd adds an already verified signature.
+	// return (true, nil) means the signature has been added
+	// return (false, nil) means the signature is a duplication
+	TrustedAdd(signerID flow.Identifier, sig crypto.Signature) (bool, error)
+	// Aggregate assumes enough shares have been collected, it aggregates the signatures
+	// and return the aggregated signature.
+	// if called concurrently, only one thread will be running the aggregation.
+	Aggregate() ([]byte, error)
+}
+
+// SignatureAggregator aggregates the verified signatures.
+// It can either be used to aggregate staking signatures or aggregate random beacon signatures,
+// but not a mix of staking signature and random beacon signature.
+type SignatureAggregator interface {
+	// TrustedAdd adds an already verified signature.
+	// return (true, nil) means the signature has been added
+	// return (false, nil) means the signature is a duplication
+	TrustedAdd(signerID flow.Identifier, sig crypto.Signature) (bool, error)
```

Comment:
I think we are missing a getter here to report how much weight we have or an extra return value for `TrustedAdd` to report weight